Kanban4U/GitHub/GitHubLogic.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+using Windows.Security.Authentication.Web;
+using Windows.UI.Xaml.Controls;
+using Windows.Web.Http;
+using Windows.Storage.Streams;
+using Windows.Web.Http.Headers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Threading;
+using Windows.UI;
+using Windows.UI.Xaml.Markup;
+
+namespace Kanban4U
+{
+    public static class GitHubLogic
+    {
+        public static async Task<string> GetGitHubAccessToken()
+        {
+            var settings = GlobalSettings.Instance;
+            if (settings.GitHubAccessToken == null)
+            {
+                settings.GitHubAccessToken = await GetGitHubAccessTokenWorker();
+            }
+
+            return settings.GitHubAccessToken;
+        }
+
+        private static SemaphoreSlim s_authenticationLock = new SemaphoreSlim(1);
+
+        public static async Task<string> GetGitHubAccessTokenWorker()
+        {
+            await s_authenticationLock.WaitAsync();
+
+            try
+            {
+                string accessCode = "";
+                WebView webView = new WebView();
+                ContentDialog dialog = new ContentDialog();
+                //dialog.Content = "Do you want to authenticate to GitHub?";
+                //dialog.PrimaryButtonText = "Yes";
+                //dialog.PrimaryButtonClick += (sender, args) => { yesClicked = true; };
+                //dialog.SecondaryButtonText = "No";
+                dialog.Content = webView;
+
+                webView.MinWidth = 400;
+                webView.MinHeight = 700;
+
+                //var msappUri = WebAuthenticationBroker.GetCurrentApplicationCallbackUri();
+                //System.Diagnostics.Debug.Assert(msappUri == new Uri("ms-app://s-1-15-2-3873301497-715253160-2862221639-955569770-3342954887-1966562334-537999965/"),
+                //    "If app callback URI changes you need to update the Azure App Registration callback URIs as well");
+                var redirectUri = new Uri("https://github.com/chrisglein/kanban4u");
```

Comment:
Will this need to be changed to the github repo you're trying to connect?

Reply:
No, that's because of how I have the GitHub OAuth application configured.